# Chapter 4 The Standard Deviation and Other Measures of Dispersion

## The Range

1. Find the range of the sets (a) 12, 6, 7, 3, 15, 10, 18, 5 and (b) 9, 3, 8, 8, 9, 8, 9, 18

In [1]:
p1.a <- c(12, 6, 7, 3, 15, 10, 18, 5)
p1.b <- c(9, 3, 8, 8, 9, 8, 9, 18)
(p1.a.range <- diff(range(p1.a)))
(p1.b.range <- diff(range(p1.b)))

[1] 15

[1] 15

---

2. Find the range of heights of the students at XYZ University. 

In [2]:
p2.heights <- c(
    "60-62",
    "63-65",
    "66-68",
    "69-71",
    "72-74"
)
p2.number.of.students <- c(5, 18, 42, 27, 8)
(p2.df <- data.frame(p2.heights, p2.number.of.students))

p2.heights,p2.number.of.students
<chr>,<dbl>
60-62,5
63-65,18
66-68,42
69-71,27
72-74,8


In [3]:
# First method - use class marks
# This method tends to eliminate extreme cases to some extent
p2.marks <- c(61, 64, 67, 70, 73)
(p2.xyz.mark.range <- diff(range(p2.marks)))

[1] 12

In [4]:
# Second method - use class boundaries
p2.boundaries <- c(59.5, 62.5, 65.5, 68.5, 71.5, 74.5)
(p2.xyz.boundary.range <- diff(range(p2.boundaries)))

[1] 15

## The mean deviation

$$
\begin{align*}
\text{MD} = \frac{\sum{|X - \bar{X}|}}{N}
\end{align*}
$$

In [5]:
md <- function(x) {
    return (sum(abs(x - mean(x)))/length(x))
}

3. Find the mean deviation of the sets of numbers in problem 1.

In [6]:
p3.x <- c(12, 6, 7, 3, 15, 10, 18, 5)
p3.y <- c(9, 3, 8, 8, 9, 8, 9, 18)
(p4.MD.x <- md(p3.x))
(p4.MD.y <- md(p3.y))

[1] 4.25

[1] 2.25

4. Find the mean deviation of the heights of the 100 male students at XYZ University

In [7]:
p4.height.classes <- c(
    "60-62",
    "63-65",
    "66-68",
    "69-71",
    "72-74"
)
p4.f <- number.of.students <- c(5, 18, 42, 27, 8)
p4.N <- sum(p4.f)
p4.X <- marks <- c(61, 64, 67, 70, 73)
p4.A <- guess <- 67
p4.d <- p4.X - p4.A
p4.X.bar <- p4.avg <- p4.A + sum(p4.f*p4.d)/p4.N
p4.dev <- abs(p4.X - p4.X.bar)
p4.fX <- p4.f*p4.dev
p4.df <- data.frame(p4.height.classes, p4.X, p4.dev, p4.f, p4.fX)
p4.fX.sum <- sum(p4.fX)
(p4.MD.xyz <- signif(p4.fX.sum/p4.N, 4))
p4.MD.xyz <- 2.26
p4.boundaries <- c(59.5, 62.5, 65.5, 68.5, 71.5, 74.5)

[1] 2.265

5. Determine the percentage of the students' heights in Problem 4 that fall within the ranges (a) $\bar{X}\pm\text{MD}$, (b) $\bar{X}\pm2\text{MD}$, and (c) $\bar{X}\pm3\text{MD}$.

In [8]:
p5.r1 <- round(range(p4.X.bar-p4.MD.xyz, p4.X.bar+p4.MD.xyz),2)
p5.r2 <- round(range(p4.X.bar-2*p4.MD.xyz, p4.X.bar+2*p4.MD.xyz),2)
p5.r3 <- round(range(p4.X.bar-3*p4.MD.xyz, p4.X.bar+3*p4.MD.xyz),2)
(round(p5.r1.pct <- 42 + (p4.boundaries[3]-p5.r1[1])/3*18 + (p5.r1[2]-p4.boundaries[4])/3*27,0))
(round(p5.r2.pct <- 18 - (p5.r2[1]-p4.boundaries[2])/3*18 + 42 + 27 + (p5.r2[2]-p4.boundaries[5])/3*8,0))
(round(p5.r3.pct <- 5 - (p5.r3[1] - p4.boundaries[1])/3*5 + 18 + 42 + 27 + (p4.boundaries[6] - p5.r3[2])/3*8,0))

[1] 55

[1] 86

[1] 91

---

## The Semi-Interquartile Range

$$
\begin{align*}
Q = \frac{Q_3 - Q_1}{2}
\end{align*}
$$

6. Find the semi-interquartile range for the height distribution of the students at XYZ University.

In [9]:
#load("Saves/Table_2.1.Rdata")
p6.Q1 <- round(65.5 + 2/42*3,2)
p6.Q3 <- round(68.5 + 10/27*3,2)
(p6.Q <- (p6.Q3 - p6.Q1)/2)

[1] 1.985

---

7. Find the semi-interquartile range for the wages of the $65$ employees at the P&R Company in Table 2.6.

In [10]:
#load("Saves/Table_2.6.Rdata")
# From Chapter 3 problem 44
p7.Q1 <- round(259.995 + 8.25/10*10,2)
p7.Q3 <- round(289.995 + .75/10*10,2)
(p7.Q <- (p7.Q3 - p7.Q1)/2)

[1] 11.25

---

## The 10-90 Percentile Range

8. Find the $10$&mdash;$90$ percentile range of the heights of the students at XYZ University (Table 2.1).

In [11]:
#load("Saves/Table_2.1.Rdata")
p8.P10 <- 62.5 + 5/18*3
p8.P90 <- 68.5 + 25/27*3
round(p8.P90 - p8.P10, 2)
(p8.80.pct.mid <- round((p8.P10 + p8.P90)/2,2))
(p8.80.pct.interval <- round((p8.P90 - p8.P10)/2,2))

[1] 7.94

[1] 67.31

[1] 3.97

Since $\frac{1}{2}(P_{10}+P_{90}) = 67.30$in and $\frac{1}{2}(P_{90}-P_{10}) = 3.97$in, we can conclude that $80$% of the students have heights in the range $67.30\pm3.97$in.

---

## The Standard Deviation

$$
\begin{align*}
s
= \sqrt{\frac{\sum^N_{j=1}{(X_j - \bar{X})^2}}{N}}
= \sqrt{\frac{\sum{(X=\bar{X})^2}}{N}}
= \sqrt{\frac{\sum{x^2}}{N}}
= \sqrt{(X - \bar{X})^2}
\end{align*}
$$

In [30]:
sd <- function(x) {
        return (sqrt(sum((x - mean(x))^2)/length(x)))
}

9. Find the standard deviation $s$ of each set of numbers in Problem 1.

In [33]:
round(sd(p1.a),2)
round(sd(p1.b),2)

[1] 4.87

[1] 3.87

---

10. Standard deviation formulas.

__When should the SD be computed with a denominator of n?__

Statistics books often show two equations to compute the SD, one using n, and the other using n-1, in the denominator. Some calculators have two buttons. 

The n-1 equation is used in the common situation where you are analyzing a sample of data and wish to make more general conclusions. The SD computed this way (with n-1 in the denominator) is your best guess for the value of the SD in the overall population.

If you simply want to quantify the variation in a particular set of data, and don't plan to extrapolate to make wider conclusions, then you can compute the SD using n in the denominator. The resulting SD is the SD of those particular values. It makes no sense to compute the SD this way if you want to estimate the SD of the population from which those points were drawn. It only makes sense to use n in the denominator when there is no sampling from a population, there is no desire to make general conclusions. 

The goal of science is always to generalize, so the equation with n in the denominator should not be used. The only example I can think of where it might make sense is in quantifying the variation among exam scores. But much better would be to show a scatterplot of every score, or a frequency distribution histogram. 

GraphPad Prism and InStat always compute the SD with the n-1 denominator.

---

11. Find the standard deviation of the heights of the 100 male students at XYZ University (Table 2.1).

In [48]:
load("Saves/Table_2.1.Rdata")
p11.f <- as.vector(Table_2.1)
p11.X <- c(61, 64, 67, 70,73)
p11.X.bar <- sum(p11.f*p11.X)/sum(p11.f)
(p11.s <- round(sqrt(sum(p11.f*(p11.X-p11.X.bar)^2)/sum(p11.f)),2))

[1] 2.92